File: api/Enkl.Api/Controllers/TaskCommentsController.cs
using Enkl.Api.Auth;
using Enkl.Api.Dtos;
using Enkl.Api.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace Enkl.Api.Controllers;

[ApiController]
[Authorize(Policy = "ProjectMember")]
[Route("api/projects/{projectId:guid}/tasks/{taskId:guid}/comments")]
public class TaskCommentsController : ControllerBase
{
    private readonly TaskCommentService _comments;

    public TaskCommentsController(TaskCommentService comments)
    {
        _comments = comments;
    }

    [HttpPost]
    public async Task<IActionResult> Create(Guid projectId, Guid taskId, CreateTaskCommentRequest request)
    {
        var result = await _comments.CreateAsync(User.UserId(), projectId, taskId, request);
        return result is null ? NotFound() : Ok(result);
    }

    [HttpPut("{commentId:guid}")]
    public async Task<IActionResult> Update(Guid projectId, Guid taskId, Guid commentId, UpdateTaskCommentRequest request)
    {
        var result = await _comments.UpdateAsync(User.UserId(), projectId, taskId, commentId, request);
        return result is null ? NotFound() : Ok(result);
    }

    [HttpDelete("{commentId:guid}")]
    public async Task<IActionResult> Delete(Guid projectId, Guid taskId, Guid commentId)
    {
        var callerClaimsOrgAdmin = User.HasClaim("orgAdmin", "true");
        var deleted = await _comments.DeleteAsync(User.UserId(), projectId, taskId, commentId, callerClaimsOrgAdmin, User.TryOrgId());
        return deleted ? NoContent() : NotFound();
    }
}

File: php-api/src/Controllers/TaskCommentsController.php
<?php

declare(strict_types=1);

namespace Enkl\Api\Controllers;

use Enkl\Api\Db\Database;
use Enkl\Api\Services\TaskCommentService;
use Psr\Http\Message\ResponseInterface as Response;
use Psr\Http\Message\ServerRequestInterface as Request;

/** Ported from Controllers/TaskCommentsController.cs. */
final class TaskCommentsController extends BaseController
{
    private function service(): TaskCommentService
    {
        return new TaskCommentService(Database::connection());
    }

    public function create(Request $request, Response $response, array $args): Response
    {
        $result = $this->service()->create($args['projectId'], $args['taskId'], $this->callerUserId($request), $this->body($request));
        return $result === null ? $this->notFound($response) : $this->json($response, $result);
    }

    public function update(Request $request, Response $response, array $args): Response
    {
        $result = $this->service()->update($args['projectId'], $args['taskId'], $args['commentId'], $this->callerUserId($request), $this->body($request));
        return $result === null ? $this->notFound($response) : $this->json($response, $result);
    }

    public function delete(Request $request, Response $response, array $args): Response
    {
        $claims = $request->getAttribute('jwtClaims');
        $callerClaimsOrgAdmin = ($claims->orgAdmin ?? null) === 'true';
        $callerOrgId = isset($claims->orgId) ? (string) $claims->orgId : null;

        $deleted = $this->service()->delete(
            $args['projectId'], $args['taskId'], $args['commentId'],
            $this->callerUserId($request), $callerClaimsOrgAdmin, $callerOrgId
        );
        return $deleted ? $this->noContent($response) : $this->notFound($response);
    }
}
